src/Entity/TestLocation/WeeklyWorkingHours/WeeklyWorkingHours.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LML\SDK\Entity\TestLocation\WeeklyWorkingHours;
+
+use Traversable;
+use IteratorAggregate;
+use LML\SDK\Entity\TestLocation\WorkingHours\WorkingHoursInterface;
+
+/**
+ * @implements IteratorAggregate<WeeklyHoursPeriod>
+ */
+class WeeklyWorkingHours implements IteratorAggregate
+{
+    /**
+     * @param list<WorkingHoursInterface> $workingDays
+     */
+    public function __construct(
+        private array $workingDays,
+    ) {
+    }
+
+    /**
+     * @param list<WorkingHoursInterface> $workingDays
+     *
+     * @return list<WeeklyHoursPeriod>
+     */
+    public function getPeriods(): array
+    {
+        $workingDays = $this->workingDays;
+        usort(
+            $workingDays,
+            fn(
+                WorkingHoursInterface $workingDayOne,
+                WorkingHoursInterface $workingDayTwo
+            ) => $workingDayOne->getDayOfWeek()->value > $workingDayTwo->getDayOfWeek()->value ? 1 : 0
+        );
+
+        $workingHourPeriods = [];
+        $periodStartDay = null;
+
+        foreach ($workingDays as $workingDay) {
+            $nextDay = next($workingDays);
```

Comment:
Moving internal array cursor is confusing and can make BC problems in future; there has been discussions in PHP internals about that and Hacklang doesn't even allow it.

So try this instead:

```php
foreach($workingDays as $key => $workingDay) {
    $nextDay = $workingDays[$key + 1] ?? null;
}

```